MapboxMobileEvents/MMEEventsManager.m

```diff
@@ -187,6 +187,14 @@ - (void)pauseOrResumeMetricsCollectionIfRequired {
 
 - (void)flush {
     if (self.paused) {
+        [self pushDebugEventWithAttributes:@{MMEDebugEventType: MMEDebugEventTypeFlush,
+                                             MMEEventKeyLocalDebugDescription: @"Aborting flushing of event queue because collection is paused."}];
+        
+        if (self.eventQueue.count >= self.configuration.eventFlushCountThreshold) {
+            [self.eventQueue removeAllObjects];
```

Comment:
removing all objects sounds harsh, and I don't think this is the right place to do it.

Instead, we should probably evict the oldest one when a new one is added in `pushEvent`

Reply:
or, only flush up to the maximum allowed when flushing, and let the eventQueue pile up